lambda-functions/cleanup-unused-images/main.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	log "github.com/sirupsen/logrus"
+)
+
+func main() {
+	log.SetLevel(log.DebugLevel)
+
+	lambda.Start(handler)
+}
+
+func handler() {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(os.Getenv("AWS_REGION"))},
+	)
+	if err != nil {
+		log.WithError(err).Error("AWS Session failed.")
+		return
+	}
+	svc := ec2.New(sess)
+	uniqueUsedImages, err := getUniqueUsedImages(svc)
+	err = deleteAMIs(svc, uniqueUsedImages)
+
+	if err != nil {
+		log.WithError(err).Error("Failed to delete AMIs.")
+		return
+	}
+}
+
+func deleteAMIs(svc *ec2.EC2, uniqueUsedImages []string) error {
+	imagesInput := &ec2.DescribeImagesInput{
+		Owners: []*string{
+			aws.String(os.Getenv("OWNER_ID")),
+		},
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("tag:Name"),
+				Values: []*string{aws.String("mattermost-cloud-*")},
+			},
+		},
+	}
+	snapshots, err := getAllSnapshots(os.Getenv("OWNER_ID"), svc)
+	if err != nil {
+		log.WithError(err).Error("Failed to get Snapshots.")
```

Comment:
I think we should replace log.WithError with just return errors.Wrap(err, "message"). Its better to handle the error in a single place which is the handler function. Same for other cases below.

Reply:
In all places or just here @stylianosrigas ?